components/layout/query.rs

```diff
@@ -228,6 +214,39 @@ pub fn process_resolved_style_request(
             _ => return computed_style(Some(fragment)),
         };
 
+        // For line height, the resolved value is the computed value if it
+        // is "normal" and the used value otherwise.
+        if longhand_id == LonghandId::LineHeight {
+            let font = style.get_font();
+            let font_size = font.font_size.computed_size();
+            return match font.line_height {
+                // There could be a fragment, but it's only interesting for `min-width` and `min-height`,
+                // so just pass None.
+                LineHeight::Normal => computed_style(None),
+                LineHeight::Number(value) => (font_size * value.0).to_css_string(),
+                LineHeight::Length(value) => value.0.to_css_string(),
+                // For a non-block level fragment we need to find the containing block of it.
+                LineHeight::MozBlockHeight => {
+                    if style.clone_display().outside() == DisplayOutside::Inline {
+                        if let Some(parent_node) = node.parent_node() {
```

Comment:
I guess this should use `cumulative_containing_block_rect` to get the actual containing block.

Reply:
Sure, I used the `cumulative_containing_block_rect` for inline box.